sdl2promela/sdl/model.py

```diff
@@ -1715,6 +1768,73 @@ def __gather_inputs(self):
                         VariableReference(param) for param in input_block.parameters
                     ]
                     trigger.transitions[target] = block
+        self.__gather_observer_inputs()
+
+    def __gather_observer_inputs(self):
+        for state, signals in self.source.cs_mapping.items():
+            for signal in signals:
+                if signal.observer_input is not None:
+                    continue
+                if signal.inputString.startswith(
+                    "input"
+                ) or signal.inputString.startswith("output"):
+                    self.__parse_observer_input(signal, state)
+
+    def __parse_observer_input(self, signal: ogAST.ContinuousSignal, state: str):
+        info, parameter = _parseObserverInput(signal.inputString)
+        self.observer_attachments.append(info)
+        parameter_type = None
+        if info.recipientName is not None:
+            type_name = "{}-Event-msg-in".format(info.recipientName.capitalize())
+            if type_name not in self.types:
+                raise Exception("Something is broken")
+            type = self.types[type_name].type
+            candidates = [
+                elem
+                for elem in type.Children.keys()
+                if elem.lower() == info.originalSignalName.lower()
+            ]
+            if len(candidates) != 1:
+                raise Exception("Something is broken")
```

Comment:
I would prefer a more informative message.

Reply:
Also in other places...

Reply:
In fact, the code seems repeated, so extracting it will enable us to fix this message only in one place

Reply:
Refactored, messge was changed